languages/python/oso/tests/test_polar.py

```diff
@@ -377,14 +381,14 @@ def test_parser_errors(polar):
         "'\\n' is not a valid character. Found in this is not at line 2, column 29"
     )
 
+    # TODO(gj): figure out what changed.
     rules = """
-    f(a) if a = "this is not allowed\0
-    """
+    f(a) if a = "this is not allowed\0"""
 
     with pytest.raises(exceptions.InvalidTokenCharacter) as e:
         polar.load_str(rules)
     assert str(e.value).startswith(
-        "'\\u{0}' is not a valid character. Found in this is not allowed at line 2, column 17"
+        "'\\u{0}' is not a valid character. Found in this is not allowed\\u{0} at line 2, column 17"
```

Comment:
The change you're referring to is that you removed the new line character after the end of file character? I'm not sure what would cause this but that sounds right.

Reply:
> The change you're referring to is that you removed the new line character after the end of file character? I'm not sure what would cause this but that sounds right.

Yeah, if I left the test string unchanged (`f(a) if a = "this is not allowed\0\n`), then the error would indicate that `'\\n'` (the final character) was the invalid character, whereas before this PR the error indicated that `'\\u{0}'` (the penultimate character) was the invalid character.